Server crashes when attempting to view non-existant profile

When logged in (as either a normal user, or an admin), enter a URL for a profile with an invalid user ID, such as `localhost:1337/profile/5` (for a database with less than 5 users). This will send a request to `localhost:1337/api/user/5`, which causes `.findOne()` to return NULL, and eventually causes a null error [here](https://github.com/18F/midas/blob/devel/api/services/utils/user.js#L760).

*edit: the `/tasks/:id` route also suffers from this problem. All calls to `.findOne()` must be null checked. The `/project/:id` route [already performs this check](https://github.com/18F/midas/blob/devel/api/services/utils/project.js#L12)*

The problem does not occur with a logged out user, since the authentication policy prevents the request from continuing.

*edit: see PR #849 for a possible solution*

```
2015-06-15T15:47:36.338Z - debug: --------------------------------------------------------
2015-06-15T15:47:36.339Z - debug: :: Mon Jun 15 2015 11:47:36 GMT-0400 (EDT)
2015-06-15T15:47:36.340Z - debug: Environment : production
2015-06-15T15:47:36.340Z - debug: Port        : 1337
2015-06-15T15:47:36.342Z - debug: --------------------------------------------------------
2015-06-15T15:48:10.762Z - debug: User Found: { username: 'bcw7044@rit.edu',
  name: 'Brendan Whitfield',
  title: null,
  bio: null,
  photoId: null,
  photoUrl: null,
  isAdmin: false,
  disabled: false,
  passwordAttempts: 0,
  id: 2,
  createdAt: '2015-06-15T15:39:00.000Z',
  updatedAt: '2015-06-15T15:39:00.000Z' }

/home/brendan/midas/api/services/utils/user.js:760
      delete user.deletedAt;
             ^
TypeError: Cannot convert null to object
    at /home/brendan/midas/api/services/utils/user.js:760:14
    at wrapper (/home/brendan/midas/node_modules/sails/node_modules/waterline/node_modules/lodash/index.js:3602:19)
    at applyInOriginalCtx (/home/brendan/midas/node_modules/sails/node_modules/waterline/lib/waterline/utils/normalize.js:421:80)
    at wrappedCallback (/home/brendan/midas/node_modules/sails/node_modules/waterline/lib/waterline/utils/normalize.js:320:18)
    at _normalizeCallback.callback.success (/home/brendan/midas/node_modules/sails/node_modules/waterline/node_modules/switchback/lib/normalize.js:33:31)
    at _switch (/home/brendan/midas/node_modules/sails/node_modules/waterline/node_modules/switchback/lib/factory.js:48:28)
    at returnResults (/home/brendan/midas/node_modules/sails/node_modules/waterline/lib/waterline/query/finders/basic.js:170:9)
    at /home/brendan/midas/node_modules/sails/node_modules/waterline/lib/waterline/query/finders/basic.js:81:16
    at /home/brendan/midas/node_modules/sails/node_modules/waterline/lib/waterline/query/finders/operations.js:77:45
    at wrapper (/home/brendan/midas/node_modules/sails/node_modules/waterline/node_modules/lodash/index.js:3602:19)
    at applyInOriginalCtx (/home/brendan/midas/node_modules/sails/node_modules/waterline/lib/waterline/utils/normalize.js:421:80)
    at wrappedCallback (/home/brendan/midas/node_modules/sails/node_modules/waterline/lib/waterline/utils/normalize.js:320:18)
    at _normalizeCallback.callback.success (/home/brendan/midas/node_modules/sails/node_modules/waterline/node_modules/switchback/lib/normalize.js:33:31)
    at _switch (/home/brendan/midas/node_modules/sails/node_modules/waterline/node_modules/switchback/lib/factory.js:48:28)
    at /home/brendan/midas/node_modules/sails-postgresql/lib/adapter.js:1162:16
    at Cursor.attachRecords (/home/brendan/midas/node_modules/sails-postgresql/node_modules/waterline-cursor/cursor/cursor.js:78:12)
npm ERR! weird error 8
npm ERR! not ok code 0
```

--- a/api/controllers/UserController.js
+++ b/api/controllers/UserController.js
@@ -69,8 +69,8 @@ module.exports = {
     }
     sails.services.utils.user.getUser(userId, reqId, req.user, function (err, user) {
       // this will only be shown to logged in users.
-      if (userId !== reqId) user.username = null;
       if (err) { return res.send(400, err); }
+      if (userId !== reqId) user.username = null;
       sails.log.debug('User Get:', user);
       res.send(user);
     });

--- a/api/services/utils/task.js
+++ b/api/services/utils/task.js
@@ -15,6 +15,7 @@ var authorized = function (id, userId, user, cb) {
   }
   Task.findOneById(id).populate('tags').exec(function (err, task) {
     if (err) { return cb('Error finding task.', null); }
+    if (!task) { return cb('Task not found.', null); }
     task.isOwner = false;
     // otherwise, check that we have an owner
     if (userId && (userId == task.userId)) {

--- a/api/services/utils/task.js
+++ b/api/services/utils/task.js
@@ -14,8 +14,7 @@ var authorized = function (id, userId, user, cb) {
     user = undefined;
   }
   Task.findOneById(id).populate('tags').exec(function (err, task) {
-    if (err) { return cb('Error finding task.', null); }
-    if (!task) { return cb('Task not found.', null); }
+    if (err || !task) { return cb('Error finding task.', null); }
     task.isOwner = false;
     // otherwise, check that we have an owner
     if (userId && (userId == task.userId)) {

--- a/api/services/utils/user.js
+++ b/api/services/utils/user.js
@@ -756,8 +756,7 @@ module.exports = {
       reqUser = undefined;
     }
     User.findOne({ id: userId }).populate('tags').exec(function (err, user) {
-      if (err) { return cb(err, null); }
-      if (!user) { return cb("User not found", null); }
+      if (err || !user) { return cb("Error finding User.", null); }
       delete user.deletedAt;
       if (userId != reqId) {
         user = self.cleanUser(user, reqId);

--- a/api/services/utils/user.js
+++ b/api/services/utils/user.js
@@ -757,6 +757,7 @@ module.exports = {
     }
     User.findOne({ id: userId }).populate('tags').exec(function (err, user) {
       if (err) { return cb(err, null); }
+      if (!user) { return cb("User not found", null); }
       delete user.deletedAt;
       if (userId != reqId) {
         user = self.cleanUser(user, reqId);

